hydromt_delft3dfm/dflowfm.py

```diff
@@ -1487,6 +1458,385 @@ def setup_1dboundary(
             da_out, name=f"boundary1d_{da_out.name}_{branch_type}"
         )  # FIXME: this format cannot be read back due to lack of branch type info from model files
 
+    def _read_forcing_geodataset(
+        self, forcing_geodataset_fn: Union[str, Path], forcing_name: str = "discharge"
+    ):
+        """Read forcing geodataset."""
+
+        refdate, tstart, tstop = self.get_model_time()  # time slice
+
+        if (
+            forcing_geodataset_fn is not None
+            and self.data_catalog[forcing_geodataset_fn].data_type == "GeoDataset"
+        ):
+            da = self.data_catalog.get_geodataset(
+                forcing_geodataset_fn,
+                geom=self.region.buffer(
+                    1000
+                ),  # only select data within region of interest (large region for forcing)
+                variables=[forcing_name],
+                time_tuple=(tstart, tstop),
+            ).rename(forcing_name)
+            # error if time mismatch
+            if np.logical_and(
+                pd.to_datetime(da.time.values[0]) == pd.to_datetime(tstart),
+                pd.to_datetime(da.time.values[-1]) == pd.to_datetime(tstop),
+            ):
+                pass
+            else:
+                self.logger.error(
+                    "forcing has different start and end time. Please check the forcing file. support yyyy-mm-dd HH:MM:SS. "
+                )
+            # reproject if needed and convert to location
+            if da.vector.crs != self.crs:
+                da = da.vector.to_crs(self.crs)
+            # get geom
+            gdf = da.vector.to_gdf(reducer=np.mean)
+        elif (
+            forcing_geodataset_fn is not None
+            and self.data_catalog[forcing_geodataset_fn].data_type == "GeoDataFrame"
+        ):
+            gdf = self.data_catalog.get_geodataframe(
+                forcing_geodataset_fn,
+                geom=self.region.buffer(self._network_snap_offset),
+            )
+            da = None
+        else:
+            gdf = None
+            da = None
+        return gdf, da
+
+    def _snap_geom_to_branches_and_drop_nonsnapped(
+        self, branches: gpd.GeoDataFrame, geoms: gpd.GeoDataFrame, snap_offset=0.0
+    ):
+        """Snaps geoms to branches and drop the ones that are not snapped.
+        Returns snapped geoms with branchid and chainage.
+        branches must have branchid.
+        """
+        workflows.find_nearest_branch(
+            branches=branches,
+            geometries=geoms,
+            maxdist=snap_offset,
+        )
+        geoms = geoms.rename(
+            columns={"branch_id": "branchid", "branch_offset": "chainage"}
+        )
+
+        # drop ones non snapped
+        _drop_geoms = geoms["chainage"].isna()
+        if any(_drop_geoms):
+            self.logger.debug(f"Unable to snap to branches: {geoms[_drop_geoms].index}")
+
+        return geoms[~_drop_geoms]
+
+    def setup_1dlateral_from_points(
```

Comment:
Same remark than for boundary, maybe split in two functions for constant vs timeseries. In that case lateral_value argument could be renamed to fillvalue to make it clear you would fill nodata value in timeseries. You could also set it to None for no filling so that maybe you could more easily spot problems in input dataset?

Reply:
a seperate issue will be created.